godef.go

```diff
@@ -225,7 +227,23 @@ func (o orderedObjects) Swap(i, j int)      { o[i], o[j] = o[j], o[i] }
 func done(obj *ast.Object, typ types.Type) {
 	defer os.Exit(0)
 	pos := types.FileSet.Position(types.DeclPos(obj))
-	fmt.Printf("%v\n", pos)
+	if *jsonFlag {
+		jsonMap := make(map[string]string)
+		if pos.Filename != "" {
+			jsonMap["filename"] = pos.Filename
+			if pos.IsValid() {
+				jsonMap["line"] = strconv.Itoa(pos.Line)
+				jsonMap["column"] = strconv.Itoa(pos.Column)
+			}
+		}
+		jsonStr, err := json.Marshal(jsonMap)
```

Comment:
If you don't like the caps (fair enough), then just define the struct
with lower case names:
     var p  = struct {
          Filename string `json:"filename,omitempty"`
          Line int `json:"line,omitempty"`
          Column int `json:"column,omitempty"`
     }{
           Filename: pos.Filename,
     etc

No need for a map.